s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/query/QueryEnding.java

```diff
@@ -0,0 +1,42 @@
+package pl.cyfronet.s4e.data.repository.query;
+
+import java.util.List;
+import java.util.Map;
+
+public class QueryEnding extends QueryDecorator {
+    public QueryEnding(QueryBuilder queryBuilder) {
+        super(queryBuilder);
+    }
+
+    public StringBuilder prepareQueryAndParameters(Map<String, Object> params, List<Object> parameters, StringBuilder resultQuery) {
+        super.prepareQueryAndParameters(params, parameters, resultQuery);
```

Comment:
Wouldn't it be better to add an abstract method `doPrepareQueryAndParameters` to QueryDecorator, make `prepareQueryAndParameters` final, and call the delegate's method and then the `do` method?